### DISCUSSION TITLE
tabulate sub-cellular localizations for metabolic enzymes

### DISCUSSION THREAD

**Discussion by haowang-bioinfo** [SysBioChalmers/Human-GEM#436] [Ideas]
Here it is proposed to prepare a table of sub-cellular localisation information for metabolic enzymes by extracting annotations from external databases (e.g. [Uniprot](https://www.uniprot.org/help/subcellular_location)), with the following consideration:

- Use as a reference for adding/removing reactions
- Adjust GPR rules according to enzymes' sub-cellular localisation
- formulate in plain text file and expect to be updated regularly

**I hereby confirm that I have:**
- [ ] Tested my code on my own computer for running the model
- [X] Done this analysis in the `main` branch of the repository
- [X] Checked that a similar issue does not exist already

**Comment by JonathanRob** [comment]
I like the idea. Since this is gene-level information, would it make sense to add this location information as an additional column to the `genes.tsv` annotation file? I recall that the `genes.tsv` file was handled differently (more automatically) than the `metabolites.tsv` and `reactions.tsv` files so maybe not so straightforward, but something to consider.

**Comment by feiranl** [comment]
Then maybe two columns, one for the localization documentation, the other for the source. We have discussed that if the localization is missing, then we will use [some deep learning tools](https://academic.oup.com/nar/article/50/W1/W228/6576357) for localization prediction. Then in that sense, source can be either from the [Uniprot](https://www.uniprot.org/help/subcellular_location)/[HPA](https://www.proteinatlas.org/search/has_protein_data_in%3ACell) [with the publication of this [paper](https://www.science.org/doi/full/10.1126/science.aal3321)] or  prediction. Of course there are many other prediction methods, I suggest we just use one for now.

**Reply by mihai-sysbio** [reply]
How about something like:
| localizationUniprot | localizationHPA | localizationPredicted |
| - | - | - |

**Reply by mihai-sysbio** [reply]
Alternatively, if more localization columns are needed, the other solution would be to have a `localization` file, with all of these, and have a `final` column in it that makes a decision based on the rest of the columns. This `final` column would then be copied in the `genes.tsv` file.

**Reply by haowang-bioinfo** [reply]
I'm open to this `localization` file

**Reply by feiranl** [reply]
I am ok with both: either `localization` file or directly combined in the `gene.tsv`.  I  prefer `gene.tsv` a little bit , since in that case, we don't store identical files which exist in Uniprot or HPA database, which stands for the minimum principle.

**Reply by haowang-bioinfo** [reply]
the current implementation is to `genes.tsv`, a `localization` file could be for later optimization?

**Reply by mihai-sysbio** [reply]
Sounds good, as long as it clear what the `localization` column is based on.

**Comment by haowang-bioinfo** [comment]
> I like the idea. Since this is gene-level information, would it make sense to add this location information as an additional column to the genes.tsv annotation file? I recall that the genes.tsv file was handled differently (more automatically) than the metabolites.tsv and reactions.tsv files so maybe not so straightforward, but something to consider.

now this location information is added as additional columns to `genes.tsv` in #437 

> Then maybe two columns, one for the localization documentation, the other for the source. We have discussed that if the localization is missing, then we will use [some deep learning tools](https://academic.oup.com/nar/article/50/W1/W228/6576357) for localization prediction. Then in that sense, source can be either from the [Uniprot](https://www.uniprot.org/help/subcellular_location)/[HPA](https://www.proteinatlas.org/search/has_protein_data_in%3ACell) [with the publication of this [paper](https://www.science.org/doi/full/10.1126/science.aal3321)] or prediction.

yes two columns were added for both the location and data sources. And the Uniprot and HPA Cell Atlas information were combined when available.

**Comment by haowang-bioinfo** [comment]
I came up with the following question when combining compartment info from SwissProt and Cell Atlas.

Both data sources are reliable for subcellular location; SwissProt assigns compartment manually according to provided publications, while Cell Atlas determines the location based on immunofluorescence (IF) microscopy detection. But IF technique has limitations, specifically in differentiating between mitochondrial matrix and inner/inter membrane space. So in the case that SwissProt assigns a protein specifically to "Inner mitochondria" (without "Mitochondria"). This assignment probably should override Cell Atlas "Mitochondria" assignment. What do you guys think?

**Comment by feiranl** [comment]
I agree that more precise subcellular localization should be kept as you suggested here using inner mitochondria to override the Mitochondria assignment. As for the specific case here for `inner mitochondria`, there are only two mets in HumanGEM are annotated in this compartment: `H+` and `Pi`, so this inner mitochondria should not be an issue in this update, but for the long term, I totally agree that we should use precise subcellular localization.

Another thing is that please note that one protein can localize at different compartment, in that case, keep them all, for example, both Uniprot and HPA annotate the protein to mitochondria and cytosol.

**Comment by haowang-bioinfo** [comment]
> please note that one protein can localize at different compartment, in that case, keep them all

yes, the current implementation is generally to keep all assignments because both are reliable data sources

> I agree that more precise subcellular localization should be kept as you suggested here using inner mitochondria to override the Mitochondria assignment.

now this change has been made in 06aa71f60b014591a2e8ea7b9fe6099a017c4225, where when a protein is assigned to "inner mitochondria" (without "Mitochondria") by SwissProt it overrides the "Mitochondria" assignment by Cell Atlas.

**Comment by johan-gson** [comment]
Nice idea! I have seen that Recon3D uses isoforms (transcripts) instead of genes for GrRules. I suspect that different isoforms sometimes may be located in different compartments (different signal peptide). Does anyone have insight in this? Why have they otherwise done that in Recon3D? Maybe not for now, but may be something to think about for later?

**Comment by feiranl** [comment]
Wow! I like this idea! Currently whenI build the ecModel for Human, every time I choose the longest transcripts for kcat prediction and its MW for the enzyme usage calculation, which should not be the case. Taking into account the isoforms will solve the issue. We should consider this.

**Comment by haowang-bioinfo** [comment]
> uses isoforms (transcripts) instead of genes for GrRules

Very good idea. It would be good to know if there's any omics data that can be traced to isforms?

**Comment by johan-gson** [comment]
You can get isoform information out of normal RNA-Seq (full length, bulk or Smart-Seq single-cell), i.e., you get ENST instead of ENSG. For 3' protocols such as 10X Chromium/Visium, you normally cannot, since the reads end up close to the 3-end of the transcripts, and hence do not cover all exomes. What I'm thinking is that it would be good both for different types of predictions (as Feiran said) or in for example ftINIT, I'm thinking it may be possible to separate reactions in different compartments. But I don't really know if it helps, it is just a guess. But we should also remember that isoform information in RNA-Seq in general is less reliable, and many isoforms are not known. Long-read technologies such as Oxford Nanopore for sure gives such information, but it is a bit noisy.

**Comment by johan-gson** [comment]
A good idea may be to have both the normal GrRules and and isoform GrRules field - when you don't want the isoform information, it is a bit annoying to have to deal with it.
